firmware/application/apps/ble_app.cpp

```diff
@@ -0,0 +1,403 @@
+        case RESERVED2:
+        case RESERVED3:
+        case RESERVED4:
+        case RESERVED5:
+        case RESERVED6:
+        case RESERVED7:
+        case RESERVED8:
+            str_console += "RESERVED";
+            break;
+        default:
+            str_console += "UNKNOWN";
+            break;
+    }
+
+    // str_console += to_string_dec_uint(value);
+
+    str_console += " Len: ";
+    str_console += to_string_dec_uint(packet->size);
+
+    str_console += "\n";
+
+    str_console += "Mac";
+    str_console += ":" + to_string_hex(packet->macAddress[0], 2);
+    str_console += ":" + to_string_hex(packet->macAddress[1], 2);
+    str_console += ":" + to_string_hex(packet->macAddress[2], 2);
+    str_console += ":" + to_string_hex(packet->macAddress[3], 2);
+    str_console += ":" + to_string_hex(packet->macAddress[4], 2);
+    str_console += ":" + to_string_hex(packet->macAddress[5], 2);
+
+    str_console += "\n";
+    str_console += "Data:";
+
+    int i;
+
+    for (i = 0; i < packet->dataLen; i++) {
+        str_console += " " + to_string_hex(packet->data[i], 2);
+    }
+
+    str_console += "\n";
+
+    // Start of Packet stuffing.
+    uint64_t macAddressEncoded = 0;
+
+    memcpy(&macAddressEncoded, packet->macAddress, sizeof(uint64_t));
+
+#if USE_CONSOLE
+    str_console += "\n" + to_string_dec_uint(macAddressEncoded) + "\n";
+    console.write(str_console);
+#else
+    // Masking off the top 2 bytes to avoid invalid keys.
+    auto& entry = ::on_packet(recent, macAddressEncoded & 0xFFFFFFFFFFFF);
+
+    entry.dbValue = packet->max_dB;
+    entry.packetData.type = packet->type;
+    entry.packetData.size = packet->size;
+    entry.packetData.dataLen = packet->dataLen;
+
+    entry.packetData.macAddress[0] = packet->macAddress[0];
+    entry.packetData.macAddress[1] = packet->macAddress[1];
+    entry.packetData.macAddress[2] = packet->macAddress[2];
+    entry.packetData.macAddress[3] = packet->macAddress[3];
+    entry.packetData.macAddress[4] = packet->macAddress[4];
+    entry.packetData.macAddress[5] = packet->macAddress[5];
+
+    for (int i = 0; i < packet->dataLen; i++) {
+        entry.packetData.data[i] = packet->data[i];
+    }
+
+    // entry.update(packet);
+    recent_entries_view.set_dirty();
+#endif
+
+    // TODO: Crude hack, should be a more formal listener arrangement...
+    if (entry.key() == recent_entry_detail_view.entry().key()) {
+        recent_entry_detail_view.set_entry(entry);
+    }
+
+    // Log at End of Packet.
+    if (logger && logging) {
+        logger->log_raw_data(str_console);
+    }
+}
+
+void BLERxView::set_parent_rect(const Rect new_parent_rect) {
+    View::set_parent_rect(new_parent_rect);
+    const Rect content_rect{0, header_height, new_parent_rect.width(), new_parent_rect.height() - header_height};
+    recent_entries_view.set_parent_rect(content_rect);
+    recent_entry_detail_view.set_parent_rect(content_rect);
+}
+
+BLERxView::~BLERxView() {
+    audio::output::stop();
```

Comment:
QQ does this app "play" audio? I didn't see the audio enable call anywhere.
If you don't actually play audio, don't need this.

Reply:
No it doesn't it was left over from the old implementation. I'll remove.